### PR TITLE
Convex bounty

### PR DESCRIPTION
**Pull request by dhruvinparikh**
Fixes #???? 

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changelog entry

### CODE REVIEW

**Comment by dhruvinparikh** [comment]
@niklr The tests are failing.

**Comment by niklr** [comment]
Tests are failing because you changed the fork block number. You can fix it by updating the whales accordingly. However, this will most likely happen everytime the fork block number gets changed unless you find a better solution to handle whales.

**Comment by dhruvinparikh** [comment]
> Tests are failing because you changed the fork block number. You can fix it by updating the whales accordingly. However, this will most likely happen everytime the fork block number gets changed unless you find a better solution to handle whales.

Here is a list of pools that needs correct whale address. 
```json
{
"compound": {
    "pool": "0x32512Bee3848bfcBb7bEAf647aa697a100f3b706",
    "stakingPool": "0xf34DFF761145FF0B05e917811d488B441F33a968",
    "lpToken": "0x32512Bee3848bfcBb7bEAf647aa697a100f3b706",
    "tokens": ["0x845838DF265Dcd2c412A1Dc9e959c7d08537f8a2"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x12c2febc4f4b34320b4af07ce03b926eb31944d1"
  },
  "usdt": {
    "pool": "0xA1c3492b71938E144ad8bE4c2fB6810b01A43dD8",
    "stakingPool": "0x8B55351ea358e5Eda371575B031ee24F462d503e",
    "lpToken": "0xA1c3492b71938E144ad8bE4c2fB6810b01A43dD8",
    "tokens": ["0x9fC689CCaDa600B6DF723D9E47D84d76664a1F23"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x9ed20e8aa8a38a2ed8bec500c438575cb222d0b6"
  },
  "3pool": {
    "pool": "0x30D9410ED1D5DA1F6C8391af5338C93ab8d4035C",
    "stakingPool": "0x689440f2Ff927E1f24c72F1087E1FAF471eCe1c8",
    "lpToken": "0x30D9410ED1D5DA1F6C8391af5338C93ab8d4035C",
    "tokens": ["0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x76af586d041d6988cdba95347e2f857872524fea"
  },
 "husd": {
    "pool": "0xe4de776C0eA0974bfA39B8cbB9491091C8cDc1ff",
    "stakingPool": "0x353e489311b21355461353fEC2d02B73EF0eDe7f",
    "lpToken": "0xe4de776C0eA0974bfA39B8cbB9491091C8cDc1ff",
    "tokens": ["0x5B5CFE992AdAC0C9D48E05854B2d91C73a003858"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x5159128278bb4c4fd8d605576e247fda25f00d1e"
  },
"usdn": {
    "pool": "0x3689f325E88c2363274E5F3d44b6DaB8f9e1f524",
    "stakingPool": "0x4a2631d090e8b40bBDe245e687BF09e5e534A239",
    "lpToken": "0x3689f325E88c2363274E5F3d44b6DaB8f9e1f524",
    "tokens": ["0x4f3E8F405CF5aFC05D68142F3783bDfE13811522"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xb0dab85eb5896fa249eb448ee692d6d636c14c95"
  },
"usdp": {
    "pool": "0x7a5dC1FA2e1B10194bD2e2e9F1A224971A681444",
    "stakingPool": "0x24DfFd1949F888F91A0c8341Fc98a3F280a782a8",
    "lpToken": "0x7a5dC1FA2e1B10194bD2e2e9F1A224971A681444",
    "tokens": ["0x7Eb40E450b9655f4B3cC4259BCC731c63ff55ae6"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xdaef20ea4708fcff06204a4fe9ddf41db056ba18"
  },
,
  "tusd": {
    "pool": "0x0A2eA49EB5F9e23058deffD509D13DDd553c2A19",
    "stakingPool": "0x308b48F037AAa75406426dACFACA864ebd88eDbA",
    "lpToken": "0x0A2eA49EB5F9e23058deffD509D13DDd553c2A19",
    "tokens": ["0xEcd5e75AFb02eFa118AF914515D6521aaBd189F1"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x12c2febc4f4b34320b4af07ce03b926eb31944d1"
  },
  "frax": {
    "pool": "0xbE0F6478E0E4894CFb14f32855603A083A57c7dA",
    "stakingPool": "0xB900EF131301B307dB5eFcbed9DBb50A3e209B2e",
    "lpToken": "0xbE0F6478E0E4894CFb14f32855603A083A57c7dA",
    "tokens": ["0xd632f22692FaC7611d2AA1C0D552930D43CAEd3B"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x7eb2b0a9042f96b595aa809107cad58df944e7e8"
  },
  "lusd": {
    "pool": "0xFB9B2f06FDb404Fd3E2278E9A9edc8f252F273d0",
    "stakingPool": "0x2ad92A7aE036a038ff02B96c88de868ddf3f8190",
    "lpToken": "0xFB9B2f06FDb404Fd3E2278E9A9edc8f252F273d0",
    "tokens": ["0xEd279fDD11cA84bEef15AF5D39BB4d4bEE23F0cA"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x851a85cd01778a7c3ea0248b8747ee12b4711581"
  },
"tricrypto": {
    "pool": "0x18684099414dcEF486F4FA5b4e44e6eA53C8c554",
    "stakingPool": "0x5Edced358e6C0B435D53CC30fbE6f5f0833F404F",
    "lpToken": "0x18684099414dcEF486F4FA5b4e44e6eA53C8c554",
    "tokens": ["0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x3f7c10cbbb1ea1046a80b738b9eaf3217410c7f6"
  },
  "tricrypto2": {
    "pool": "0x903C9974aAA431A765e60bC07aF45f0A1B3b61fb",
    "stakingPool": "0x9D5C5E364D81DaB193b72db9E9BE9D8ee669B652",
    "lpToken": "0x903C9974aAA431A765e60bC07aF45f0A1B3b61fb",
    "tokens": ["0xc4AD29ba4B3c580e6D59105FFf484999997675Ff"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x9811edbafbdaaa9f58f62ed70e4633a5b85e08e4"
  }
}
```

Needless to say the test suite skips the tests of following pools

```json
{
"ypool": {
    "pool": "0x0928F6753880A03628eB0be07b77992c8af37874",
    "stakingPool": "0xd802a8351A76ED5eCd89A7502Ca615F2225A585d",
    "lpToken": "0x0928F6753880A03628eB0be07b77992c8af37874",
    "tokens": ["0xdF5e0e81Dff6FAF3A7e52BA697820c5e32D806A8"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x77d3c47876e45123c2837ba68720378af00a2c0a"
  },
"busd": {
    "pool": "0x59bB786F222d3f0f00B0dA31B799Fff80D552940",
    "stakingPool": "0x602c4cD53a715D8a7cf648540FAb0d3a2d546560",
    "lpToken": "0x59bB786F222d3f0f00B0dA31B799Fff80D552940",
    "tokens": ["0x3B3Ac5386837Dc563660FB6a0937DFAa5924333B"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xbab89cbe597876364b3bc270b1793c0e2e2aa995"
  },
"pax": {
    "pool": "0x2eA94b0d3349A284488ACF2934E494b2f58ef647",
    "stakingPool": "0xe3DaafC8C14147d5B4A7a56F0BfdED240158e51e",
    "lpToken": "0x2eA94b0d3349A284488ACF2934E494b2f58ef647",
    "tokens": ["0xD905e2eaeBe188fc92179b6350807D8bd91Db0D8"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xd56a91e60241fc39deebd8bfaefa398cc672d906"
  },
"sbtc": {
    "pool": "0xbA723E335eC2939D52a2efcA2a8199cb4CB93cC3",
    "stakingPool": "0xd727A5A6D1C7b31Ff9Db4Db4d24045B7dF0CFF93",
    "lpToken": "0xbA723E335eC2939D52a2efcA2a8199cb4CB93cC3",
    "tokens": ["0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x600b8a34ec1cfd8b8af78cfc49708419a16ea2e8"
  },
"hbtc": {
    "pool": "0x33c00bF8CFDf42929E0884d230A55F963221f8f3",
    "stakingPool": "0x618BD6cBA676a46958c63700C04318c84a7b7c0A",
    "lpToken": "0x33c00bF8CFDf42929E0884d230A55F963221f8f3",
    "tokens": ["0xb19059ebb43466C323583928285a49f558E572Fd"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x7a7a599d2384ed203cfea49721628aa851e0da16"
  },
"pbtc": {
    "pool": "0x21Cce64289407081744F087950b9DB32906470fC",
    "stakingPool": "0x2d3C90AEB11D1393CA839Afc9587515B1325D77A",
    "lpToken": "0x21Cce64289407081744F087950b9DB32906470fC",
    "tokens": ["0xDE5331AC4B3630f94853Ff322B66407e0D6331E8"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x5a87e9a0a765fe5a69fa6492d3c7838dc1511805"
  },
"bbtc": {
    "pool": "0x2E1f902b9067b5fDd7AF29ef05D4fF6212588388",
    "stakingPool": "0x61D741045cCAA5a215cF4E5e55f20E1199B4B843",
    "lpToken": "0x2E1f902b9067b5fDd7AF29ef05D4fF6212588388",
    "tokens": ["0x410e3E86ef427e30B9235497143881f717d93c2A"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x4cf2c3ce9ca9a01e78b85fa12ac0352be874aacb"
  },
 "obtc": {
    "pool": "0xc1C030139eEc070Ed8FD092CC8C273C638A18bBe",
    "stakingPool": "0xeeeCE77e0bc5e59c77fc408789A9A172A504bD2f",
    "lpToken": "0xc1C030139eEc070Ed8FD092CC8C273C638A18bBe",
    "tokens": ["0x2fE94ea3d5d4a175184081439753DE15AeF9d614"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xd41f7006bcb2b3d0f9c5873272ebed67b37f80dc"
  },
"eurs": {
    "pool": "0xd7E2b9494c529b42Dea53EF6a237C16502E6A927",
    "stakingPool": "0xcB8F69E0064d8cdD29cbEb45A14cf771D904BcD3",
    "lpToken": "0xd7E2b9494c529b42Dea53EF6a237C16502E6A927",
    "tokens": ["0x194eBd173F6cDacE046C53eACcE9B953F28411d1"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xaffe4271096c9ecbbde1bb309c60136506989acb"
  },
"seth": {
    "pool": "0xAF1d4C576bF55f6aE493AEebAcC3a227675e5B98",
    "stakingPool": "0x192469CadE297D6B21F418cFA8c366b63FFC9f9b",
    "lpToken": "0xAF1d4C576bF55f6aE493AEebAcC3a227675e5B98",
    "tokens": ["0xA3D87FffcE63B53E0d54fAa1cc983B7eB0b74A9c"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x9ea83407e0046ee452bc6535c0aa5609d7f6f680"
  },
"aave": {
    "pool": "0x23F224C37C3A69A058d86a54D3f561295A93d542",
    "stakingPool": "0xE82c1eB4BC6F92f85BF7EB6421ab3b882C3F5a7B",
    "lpToken": "0x23F224C37C3A69A058d86a54D3f561295A93d542",
    "tokens": ["0xFd2a8fA60Abd58Efe3EeE34dd494cD491dC14900"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x33d7dcfd18c9220b41f592bd2b77e70cad699577"
  },
"steth": {
    "pool": "0x9518c9063eB0262D791f38d8d6Eb0aca33c63ed0",
    "stakingPool": "0x0A760466E1B4621579a82a39CB56Dda2F4E70f03",
    "lpToken": "0x9518c9063eB0262D791f38d8d6Eb0aca33c63ed0",
    "tokens": ["0x06325440D014e39736583c165C2963BA99fAf14E"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x8835a35023c2fcf105e1f232e600385ed6db9bc6"
  },
"saave": {
    "pool": "0x09CCD0892b696AB21436e51588a7a7f8b649733d",
    "stakingPool": "0xF86AE6790654b70727dbE58BF1a863B270317fD0",
    "lpToken": "0x09CCD0892b696AB21436e51588a7a7f8b649733d",
    "tokens": ["0x02d341CcB60fAaf662bC0554d13778015d1b285C"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xe594173aaa1493665ec6a19a0d170c76eea1124a"
  },
  "ankreth": {
    "pool": "0x7E96955b66c89B931BBDAf187740Cc0fF2602F21",
    "stakingPool": "0x8798b81b0261934aa850C8de8622472bfdc143F4",
    "lpToken": "0x7E96955b66c89B931BBDAf187740Cc0fF2602F21",
    "tokens": ["0xaA17A236F2bAdc98DDc0Cf999AbB47D47Fc0A6Cf"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x569d327288d8354f16813afb01bba0e60c1301f9"
  },
"ironbank": {
    "pool": "0x912EC00eaEbf3820a9B0AC7a5E15F381A1C91f22",
    "stakingPool": "0x3E03fFF82F77073cc590b656D42FceB12E4910A8",
    "lpToken": "0x912EC00eaEbf3820a9B0AC7a5E15F381A1C91f22",
    "tokens": ["0x5282a4eF67D9C33135340fB3289cc1711c13638C"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xc037f78075b256af8405f37658ed1ebc162d0231"
  },
"alusd": {
    "pool": "0xCA3D9F45FfA69ED454E66539298709cb2dB8cA61",
    "stakingPool": "0x02E2151D4F351881017ABdF2DD2b51150841d5B3",
    "lpToken": "0xCA3D9F45FfA69ED454E66539298709cb2dB8cA61",
    "tokens": ["0x43b4FdFD4Ff969587185cDB6f0BD875c5Fc83f8c"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xd44a4999df99fb92db7cdfe7dea352a28bcedb63"
  },
"reth": {
    "pool": "0x7ADd8D0E923CB692DF6bC65d96d510f0E2fC37af",
    "stakingPool": "0x61dB6c2321f784c8fAb8d5eF80f58F27C831dCc8",
    "lpToken": "0x7ADd8D0E923CB692DF6bC65d96d510f0E2fC37af",
    "tokens": ["0x53a901d48795C58f485cBB38df08FA96a24669D5"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0x748bde1b23b36e0b4750c8709221d2f10bf2f629"
  },
"eurt": {
    "pool": "0x2b2175AC371Ec2900AC39fb87452340F65CC9895",
    "stakingPool": "0xD814BFC091111E1417a669672144aFFAA081c3CE",
    "lpToken": "0x2b2175AC371Ec2900AC39fb87452340F65CC9895",
    "tokens": ["0xFD5dB7463a3aB53fD211b4af195c5BCCC1A03890"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xa350122a590fc6c8bee981a06039436fff79c02a"
  },
  "mim": {
    "pool": "0xabB54222c2b77158CC975a2b715a3d703c256F05",
    "stakingPool": "0xFd5AbF66b003881b88567EB9Ed9c651F14Dc4771",
    "lpToken": "0xabB54222c2b77158CC975a2b715a3d703c256F05",
    "tokens": ["0x5a6A4D54456819380173272A5E8E9B9904BdF41B"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xd3474bd2f558b12e83810ec7d8af14f2d1d08137"
  },
"eurs": {
    "pool": "0xd7E2b9494c529b42Dea53EF6a237C16502E6A927",
    "stakingPool": "0xcB8F69E0064d8cdD29cbEb45A14cf771D904BcD3",
    "lpToken": "0xd7E2b9494c529b42Dea53EF6a237C16502E6A927",
    "tokens": ["0x194eBd173F6cDacE046C53eACcE9B953F28411d1"],
    "rewardTokens": ["0xD533a949740bb3306d119CC777fa900bA034cd52"],
    "whale": "0xaffe4271096c9ecbbde1bb309c60136506989acb"
  }
}
``` 

Alternatively, the wallet can be funded using bruteforce mechanism.
Sources : https://blog.euler.finance/brute-force-storage-layout-discovery-in-erc20-contracts-with-hardhat-7ff9342143ed and https://gist.github.com/banteg/6d84db1c74383a80df03628a8c25ff67

**Comment by dhruvinparikh** [comment]
Unit tests result for rest of pools

```console
Unit tests
susd
ren
gusd
usdk
musd
rsv
tbtc
dusd
ust
link
busdv2
    ConvexFinanceAdapter
      ✓ should deposit susdCrv, stake cvxsusd3Crv, claim CRV, unstake cvxsusd3Crv, and withdraw susdCrv in susd pool of Convex Finance (7640ms)
      ✓ should deposit susdCrv, stake cvxsusd3Crv, unstake cvxsusd3Crv, and withdraw susdCrv in susd pool of Convex Finance (3699ms)
      1) should deposit renCrv, stake cvxren3Crv, claim CRV, unstake cvxren3Crv, and withdraw renCrv in ren pool of Convex Finance
      2) should deposit renCrv, stake cvxren3Crv, unstake cvxren3Crv, and withdraw renCrv in ren pool of Convex Finance
      ✓ should deposit gusdCrv, stake cvxgusd3Crv, claim CRV, unstake cvxgusd3Crv, and withdraw gusdCrv in gusd pool of Convex Finance (5187ms)
      ✓ should deposit gusdCrv, stake cvxgusd3Crv, unstake cvxgusd3Crv, and withdraw gusdCrv in gusd pool of Convex Finance (1919ms)
      3) should deposit usdkCrv, stake cvxusdk3Crv, claim CRV, unstake cvxusdk3Crv, and withdraw usdkCrv in usdk pool of Convex Finance
      4) should deposit usdkCrv, stake cvxusdk3Crv, unstake cvxusdk3Crv, and withdraw usdkCrv in usdk pool of Convex Finance
      ✓ should deposit musdCrv, stake cvxmusd3Crv, claim CRV, unstake cvxmusd3Crv, and withdraw musdCrv in musd pool of Convex Finance (5975ms)
      ✓ should deposit musdCrv, stake cvxmusd3Crv, unstake cvxmusd3Crv, and withdraw musdCrv in musd pool of Convex Finance (2854ms)
      ✓ should deposit rsvCrv, stake cvxrsv3Crv, claim CRV, unstake cvxrsv3Crv, and withdraw rsvCrv in rsv pool of Convex Finance (6138ms)
      ✓ should deposit rsvCrv, stake cvxrsv3Crv, unstake cvxrsv3Crv, and withdraw rsvCrv in rsv pool of Convex Finance (3065ms)
      ✓ should deposit tbtcCrv, stake cvxtbtc3Crv, claim CRV, unstake cvxtbtc3Crv, and withdraw tbtcCrv in tbtc pool of Convex Finance (5249ms)
      ✓ should deposit tbtcCrv, stake cvxtbtc3Crv, unstake cvxtbtc3Crv, and withdraw tbtcCrv in tbtc pool of Convex Finance (2777ms)
      ✓ should deposit dusdCrv, stake cvxdusd3Crv, claim CRV, unstake cvxdusd3Crv, and withdraw dusdCrv in dusd pool of Convex Finance (5903ms)
      ✓ should deposit dusdCrv, stake cvxdusd3Crv, unstake cvxdusd3Crv, and withdraw dusdCrv in dusd pool of Convex Finance (2853ms)
      ✓ should deposit ustCrv, stake cvxust3Crv, claim CRV, unstake cvxust3Crv, and withdraw ustCrv in ust pool of Convex Finance (16555ms)
      ✓ should deposit ustCrv, stake cvxust3Crv, unstake cvxust3Crv, and withdraw ustCrv in ust pool of Convex Finance (2608ms)
      ✓ should deposit linkCrv, stake cvxlink3Crv, claim CRV, unstake cvxlink3Crv, and withdraw linkCrv in link pool of Convex Finance (7401ms)
      ✓ should deposit linkCrv, stake cvxlink3Crv, unstake cvxlink3Crv, and withdraw linkCrv in link pool of Convex Finance (2485ms)
      ✓ should deposit busdv2Crv, stake cvxbusdv23Crv, claim CRV, unstake cvxbusdv23Crv, and withdraw busdv2Crv in busdv2 pool of Convex Finance (9886ms)
      ✓ should deposit busdv2Crv, stake cvxbusdv23Crv, unstake cvxbusdv23Crv, and withdraw busdv2Crv in busdv2 pool of Convex Finance (2799ms)


  18 passing (2m)
  4 failing

  1) Unit tests
       ConvexFinanceAdapter
         should deposit renCrv, stake cvxren3Crv, claim CRV, unstake cvxren3Crv, and withdraw renCrv in ren pool of Convex Finance:

      AssertionError: expected 0 to be above 0
      + expected - actual


      at Context.<anonymous> (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:177:88)
      at Generator.next (<anonymous>)
      at fulfilled (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:24:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at runNextTicks (internal/process/task_queues.js:64:3)
      at listOnTimeout (internal/timers.js:526:9)
      at processTimers (internal/timers.js:500:7)

  2) Unit tests
       ConvexFinanceAdapter
         should deposit renCrv, stake cvxren3Crv, unstake cvxren3Crv, and withdraw renCrv in ren pool of Convex Finance:
     AssertionError: Expected "0" to be greater than 0
      at Context.<anonymous> (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:255:13)
      at Generator.next (<anonymous>)
      at fulfilled (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:24:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at runNextTicks (internal/process/task_queues.js:64:3)
      at listOnTimeout (internal/timers.js:526:9)
      at processTimers (internal/timers.js:500:7)

  3) Unit tests
       ConvexFinanceAdapter
         should deposit usdkCrv, stake cvxusdk3Crv, claim CRV, unstake cvxusdk3Crv, and withdraw usdkCrv in usdk pool of Convex Finance:
     AssertionError: Expected "0" to be greater than 0
      at Context.<anonymous> (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:120:51)
      at Generator.next (<anonymous>)
      at fulfilled (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:24:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at runNextTicks (internal/process/task_queues.js:64:3)
      at listOnTimeout (internal/timers.js:526:9)
      at processTimers (internal/timers.js:500:7)

  4) Unit tests
       ConvexFinanceAdapter
         should deposit usdkCrv, stake cvxusdk3Crv, unstake cvxusdk3Crv, and withdraw usdkCrv in usdk pool of Convex Finance:
     AssertionError: Expected "0" to be greater than 0
      at Context.<anonymous> (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:255:13)
      at Generator.next (<anonymous>)
      at fulfilled (test/1_ethereum/convex.finance/ConvexFinanceAdapter.behavior.ts:24:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at runNextTicks (internal/process/task_queues.js:64:3)
      at listOnTimeout (internal/timers.js:526:9)
      at processTimers (internal/timers.js:500:7)
```

File: contracts/1_ethereum/convex.finance/interfaces/ICurveStableSwap.sol [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.12;
+
+interface ICurveStableSwap {
```
Comment (dhruvinparikh) [inline contracts/1_ethereum/convex.finance/interfaces/ICurveStableSwap.sol]: remove if not used

File: contracts/1_ethereum/convex.finance/interfaces/ICurveRegistry.sol [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.12;
+
+interface ICurveRegistry {
```
Comment (dhruvinparikh) [inline contracts/1_ethereum/convex.finance/interfaces/ICurveRegistry.sol]: remove if not used

File: contracts/1_ethereum/convex.finance/interfaces/ICurvePoolInfo.sol [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+interface ICurvePoolInfo {
```
Comment (dhruvinparikh) [inline contracts/1_ethereum/convex.finance/interfaces/ICurvePoolInfo.sol]: remove if not used

File: contracts/1_ethereum/convex.finance/interfaces/IConvexStake.sol [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.12;
+
+// https://github.com/convex-eth/platform/blob/main/contracts/contracts/BaseRewardPool.sol
+interface IConvexStake {
```
Comment (dhruvinparikh) [inline contracts/1_ethereum/convex.finance/interfaces/IConvexStake.sol]: remove if not used

File: contracts/1_ethereum/convex.finance/interfaces/IConvexDeposit.sol [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+// https://github.com/convex-eth/platform/blob/main/contracts/contracts/Booster.sol
+interface IConvexDeposit {
```
Comment (dhruvinparikh) [inline contracts/1_ethereum/convex.finance/interfaces/IConvexDeposit.sol]: remove if not used

**Review by dhruvinparikh — COMMENTED** [review]
Changes required

File: contracts/1_ethereum/convex.finance/interfaces/IConvexDeposit.sol (hunk 5 above)
Comment (NouDaimon) [reply contracts/1_ethereum/convex.finance/interfaces/IConvexDeposit.sol]: I removed all files which are not used.

File: contracts/1_ethereum/convex.finance/interfaces/ICurveRegistry.sol (hunk 2 above)
Comment (NouDaimon) [reply contracts/1_ethereum/convex.finance/interfaces/ICurveRegistry.sol]: Removed

File: contracts/1_ethereum/convex.finance/interfaces/ICurvePoolInfo.sol (hunk 3 above)
Comment (NouDaimon) [reply contracts/1_ethereum/convex.finance/interfaces/ICurvePoolInfo.sol]: Removed

File: contracts/1_ethereum/convex.finance/interfaces/ICurveStableSwap.sol (hunk 1 above)
Comment (NouDaimon) [reply contracts/1_ethereum/convex.finance/interfaces/ICurveStableSwap.sol]: Removed

**Pull request merged**